env0/resource_environment.go

```diff
@@ -23,27 +23,147 @@ func resourceEnvironment() *schema.Resource {
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			"revision": {
+				Type:        schema.TypeString,
+				Description: "the revision the environment is to be run against",
+				Optional:    true,
+				Computed:    true,
+			},
+			"run_plan_on_pull_requests": {
+				Type:        schema.TypeBool,
+				Description: "should run terraform plan on pull requests creations",
+				Optional:    true,
+				Computed:    true,
+			},
+			"approve_plan_automatically": {
+				Type:        schema.TypeBool,
+				Description: "should deployments require manual approvals",
+				Optional:    true,
+				Computed:    true,
+			},
+			"deploy_on_push": {
+				Type:        schema.TypeBool,
+				Description: "should run terraform deploy on push events",
+				Optional:    true,
+				Computed:    true,
+			},
+			"auto_deploy_on_path_changes_only": {
+				Type:        schema.TypeBool,
+				Description: "redeploy only on path changes only",
+				Optional:    true,
+				Computed:    true,
+			},
+			"auto_deploy_by_custom_glob": {
+				Type:        schema.TypeString,
+				Description: "redeploy on file filter pattern",
+				Optional:    true,
+			},
+			"deployment_id": {
+				Type:        schema.TypeString,
+				Description: "id of the last deployment",
+				Computed:    true,
+			},
+			"configuration": {
+				Type:        schema.TypeList,
+				Description: "terraform and environment variables for the environment",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:        schema.TypeString,
+							Description: "variable name",
+							Required:    true,
+						},
+						"value": &schema.Schema{
+							Type:        schema.TypeString,
+							Description: "variable value",
+							Required:    true,
+						},
+						"type": &schema.Schema{
+							Type:        schema.TypeString,
+							Description: "variable type (allowed values are: terraform, environment)",
+							Default:     "environment",
+							Optional:    true,
+						},
+						"description": &schema.Schema{
+							Type:        schema.TypeString,
+							Description: "description for the variable",
+							Optional:    true,
+						},
+						"is_sensitive": &schema.Schema{
+							Type:        schema.TypeBool,
+							Description: "should the variable value be hidden",
+							Optional:    true,
+						},
+						"schema_type": &schema.Schema{
+							Type:        schema.TypeString,
+							Description: "the type the variable must be of",
+							Optional:    true,
+						},
+						"schema_enum": &schema.Schema{
+							Type:        schema.TypeList,
+							Description: "a list of possible variable values",
+							Optional:    true,
+							Elem: &schema.Schema{
+								Type:        schema.TypeString,
+								Description: "name to give the configuration variable",
+							},
+						},
+					},
+				},
 			},
 		},
 	}
 }
 
+var VariableTypes = map[string]client.ConfigurationVariableType{
```

Comment:
I'm guessing this should be configured under `configuration_variable` rather than here, isn't it?

Reply:
I'll throw it model, it has consts similar to this there 👍🏻

Reply:
a99278f